x-pack/plugins/fleet/common/services/is_agent_request_diagnostics_supported.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import semverGte from 'semver/functions/gte';
+
+import type { Agent } from '../types';
+
+export const MINIMUM_DIAGNOSTICS_AGENT_VERSION = '8.7.0';
+
+export function isAgentRequestDiagnosticsSupported(agent: Agent) {
+  let agentVersion: string;
+  if (typeof agent?.local_metadata?.elastic?.agent?.version === 'string') {
+    agentVersion = agent.local_metadata.elastic.agent.version;
+  } else {
+    return false;
+  }
```

Comment:
nit: could this be written with an early return to make it more readable
```suggestion
  if (typeof agent?.local_metadata?.elastic?.agent?.version !== 'string') {
    return false
  } 

  const agentVersion = agent.local_metadata.elastic.agent.version;
```